ui/static/ui/js/listing.js

```diff
@@ -220,104 +231,79 @@ define('listing',
        * @returns {jQuery.Deferred} A promise that's resolved or rejected when
        * the AJAX call completes and the rerender is triggered.
        */
-      var refreshFromAPI;
-
-      // Will be set in refreshFromAPI
-      var selectedFacets;
-      var selectedMissingFacets;
+      refreshFromAPI: function() {
+        this.setState({pageLoaded: false});
 
-      var loadResourceTab = function(resourceId) {
-        LearningResources.loader(
-          repoSlug,
-          resourceId,
-          refreshFromAPI,
-          closeLearningResourcePanel,
-          $("#tab-1")[0]
-        );
-      };
-
-      var loadXmlTab = function(resourceId) {
-        XmlPanel.loader(repoSlug, resourceId, $("#tab-2")[0]);
-      };
-
-      var loadStaticAssetsTab = function(resourceId) {
-        StaticAssets.loader(repoSlug, resourceId, $("#tab-3")[0]);
-      };
-
-      // Keep track of resource id so we can lazy load panels.
-      var currentResourceId;
-
-      var panelLoaders = {
-        "tab-1": loadResourceTab,
-        "tab-2": loadXmlTab,
-        "tab-3": loadStaticAssetsTab
-      };
-
-      var loadedPanels;
-      var openResourcePanel = function(resourceId) {
-        // Reset loaded panels
-        loadedPanels = {};
-        currentResourceId = resourceId;
-
-        // Load the resource tab
-        showTab("tab-1");
-        loadResourceTab(currentResourceId);
-        loadedPanels["tab-1"] = true;
-
-        $('.cd-panel').addClass('is-visible');
-      };
-
-      _.each(panelLoaders, function(loader, tag) {
-        $("a[href='#" + tag + "']").click(function() {
-          // If tab not already loaded, load it now
-          if (!loadedPanels[tag]) {
-            loader(currentResourceId);
-            loadedPanels[tag] = true;
-          }
-        });
-      });
-
-      refreshFromAPI = function() {
-        pageLoaded = false;
-
-        var newQuery = "?" + URI().search(queryMap).query();
-        if (newQuery === "?") {
-          // Don't put ? in URL if empty
-          History.replaceState(null, document.title, ".");
-          newQuery = "";
-        } else {
-          History.replaceState(null, document.title, newQuery);
-        }
+        var queryMap = this.makeQueryMap();
+        var newQuery = this.makeQueryString(queryMap);
 
+        // Query string for repository page is the same used for the search API
         var url = "/api/v1/repositories/" +
-          listingOptions.repoSlug + "/search/" + newQuery;
-
-        renderListingResources();
-        return $.get(url).then(function(collection) {
-          listingOptions = $.extend({}, listingOptions);
-          listingOptions.resources = collection.results;
-          listingOptions.facetCounts = collection.facet_counts;
-          selectedFacets = collection.selected_facets;
-          selectedMissingFacets = collection.selected_missing_facets;
+          this.props.repoSlug + "/search/" + newQuery;
+
+        var thiz = this;
```

Comment:
Our style guide is whatever is enforced by JSCS and JSHint, mostly the Google style. I don't think there's a preference for or against `bind` in that style guide. I like using an explicit variable for that purpose but I don't have a strong preference either way.